source/Calamari/Deployment/PackageRetention/Repositories/JsonJournalRepository.cs

```diff
@@ -58,8 +61,25 @@ void Load()
             if (File.Exists(journalPath))
             {
                 var json = File.ReadAllText(journalPath);
-                journalEntries = JsonConvert.DeserializeObject<List<JournalEntry>>(json)
-                                            .ToDictionary(entry => entry.Package, entry => entry);
+
+                if (TryParseJournal(json, out var journalContents))
+                {
+                    journalEntries = journalContents.ToDictionary(entry => entry.Package, entry => entry);
+                }
+                else
+                {
+                    var journalFileName = Path.GetFileNameWithoutExtension(journalPath);
+                    var backupJournalFileName = $"{journalFileName}_{DateTimeOffset.UtcNow:yyyyMMddTHHmmss}.json"; // eg. PackageRetentionJournal_20210101T120000.json
+
+                    log.Warn($"The existing package retention journal file {journalPath} could not be read. The file will be renamed to {backupJournalFileName}. A new journal will be created.");
+
+                    var backupJournalPath = Path.GetDirectoryName(journalPath) + Path.DirectorySeparatorChar + backupJournalFileName;
```

Comment:
This is updated since I found that the invalid journal was being copied to the directory where the exe was running which seems not very ideal.

Reply:
Ooooh good pick up.

Use Path.Combine to join these though.

Reply:
That's much nicer!